OggVorbisEncoder/Setup/Mapping.cs

```diff
@@ -18,29 +20,33 @@ public class Mapping
             if (couplingMag?.Length != couplingAng?.Length)
                 throw new ArgumentException($"{nameof(couplingMag)} and {nameof(couplingAng)} must be the same size");
 
+            SubMaps = submaps;
             ChannelMuxList = channelMuxList;
             FloorSubMap = floorSubMap;
             ResidueSubMap = residueSubMap;
+            CouplingSteps = couplingSteps;
             CouplingMag = couplingMag;
             CouplingAng = couplingAng;
         }
 
-        public int SubMaps => FloorSubMap.Length;
+        public int SubMaps { get; }
 
         public int[] ChannelMuxList { get; } // up to 256 channels in a Vorbis stream
 
         public int[] FloorSubMap { get; } // [mux] submap to floors
         public int[] ResidueSubMap { get; } // [mux] submap to residue
 
-        public int CouplingSteps => CouplingMag.Length;
+        public int CouplingSteps { get; }
 
         public int[] CouplingMag { get; }
         public int[] CouplingAng { get; }
 
         public Mapping Clone() => new Mapping(
+            SubMaps,
             ChannelMuxList.ToArray(),
             FloorSubMap.ToArray(),
             ResidueSubMap.ToArray(),
+            CouplingSteps,
             CouplingMag.ToArray(),
             CouplingAng.ToArray());
```

Comment:
Needed to add actual counts to the struct since the ResidueType.One elements use a 0-length coupling steps array.